src/settings/ServicePoints/ServicePointFormContainer.js

```diff
@@ -3,21 +3,37 @@ import React, {
   useEffect,
   useState,
 } from 'react';
+import PropTypes from 'prop-types';
 import {
   cloneDeep,
   unset,
   orderBy,
+  omit,
+  set,
 } from 'lodash';
 
 import ServicePointForm from './ServicePointForm';
+import {
+  shortTermExpiryPeriod,
+  shortTermClosedDateManagementMenu,
+  longTermClosedDateManagementMenu
+} from './constants';
 
 const ServicePointFormContainer = ({
   onSave,
   parentResources,
   initialValues: servicePoint,
   ...rest
 }) => {
-  const [initialValues, setInitialValues] = useState(servicePoint);
+  const getServicePoint = () => {
+    // remove holdShelfClosedLibraryDateManagement from servicepoint object when pickupLocation is not true
+    if (!servicePoint.pickupLocation) {
+      omit(servicePoint, 'holdShelfClosedLibraryDateManagement');
+    }
+    return servicePoint;
+  };
+
+  const [initialValues, setInitialValues] = useState(getServicePoint());
 
   useEffect(() => {
     setInitialValues(servicePoint);
```

Comment:
Should `servicePoint` here be with `holdShelfClosedLibraryDateManagement` or without?

Reply:
Thank you for the observations.. I have updated the code. Please review